### DISCUSSION TITLE
SPI flash and EEPROM dump of the ODI DFP-34x-2c2

### DISCUSSION THREAD

**Discussion by lehoangnb** [Anime4000/RTL960x#286] [Show and tell]
# Its for ODI DFP-34x-2c2 with 8MB MX25L6433F flash and 24C04 EEPROM
![PCB - Copy](https://github.com/Anime4000/RTL960x/assets/5148339/da21178b-03b4-4139-a91a-47f5a200f086)
## SPI Flash info
- Flash name: MX25L6433F SPI 8MB
- File name: ODI_DFP-34X-2C2_factory_8M_v220923.bin
- sha256: 5e3572c2c6821b8f345d6564ae547aaf734b227c678744b276d311cb97006cd7
- Both layout is running 220923 @Anime4000 modified firmware

![Flash_info](https://github.com/Anime4000/RTL960x/assets/5148339/77430201-e211-4733-87ab-d2e34c4692e2)
![MX25L6433F_Pinout](https://github.com/Anime4000/RTL960x/assets/5148339/c0b8e5af-ef7a-46bf-977d-ee72b1f6d7cf)
![Test Clips Socket - Copy](https://github.com/Anime4000/RTL960x/assets/5148339/52a13b9a-418d-4679-9100-4ba45cffe67a)
## EEPROM info
- EEPROM name: 24C04 4K (512B) I2C (SOT23 package)
- File name: ODI_DFP-34X-2C2_eeprom_4K.bin
- sha256: 6342c4bdefee80e1ca238615c3840184eaec5cdae48dc403111bb5b09f96fef0

![EEPROM_INFO](https://github.com/Anime4000/RTL960x/assets/5148339/af31f9aa-3ec5-4c44-b956-8988ebaecf81)
![EEPROM_Program_wiring - Copy](https://github.com/Anime4000/RTL960x/assets/5148339/88d63b12-09cf-4e24-8e17-89e447e47217)
![24C04-SOT23_pinout](https://github.com/Anime4000/RTL960x/assets/5148339/55a6e776-3b02-42f2-b286-b777abf566f9)

## Notice
- Use only **3.3V** level for both Flash and EEPROM
- In most case, flashing eeprom is not necessary

## Download
[ODI_DFP-34X-2C2_full_dump_v220923.zip](https://github.com/Anime4000/RTL960x/files/15269394/ODI_DFP-34X-2C2_full_dump_v220923.zip)

**Comment by Headcrabed** [comment]
Besides, eeprom need to be modified to switch to 2500basex mode for host device.

**Comment by Anime4000** [comment]
I linking this to main README.md

**Comment by duskyzaoKING** [comment]
Hi. Do you have the original .bin of the ODI? I lost access to the IP interface, and i dont wat to do UART as i have a RT809H programmer. Will flash direct the .bin with aligator clip. I will restore original firmware and then change to M114_sfp_ODI_PPPOE_210702 firmware because they do PPPoE auth. Do your firmware do PPPoE Auth ? I have a Chinese Switch Layer 3 and i am having a lot of probleming with low speed only using my switch L3. Maybe my switch have a problem with HiSGMII PHY, HiSGMII MAC and 2500Base-X? I lost access when i changed to LAN_SDS_MODE 6. As i did no know about your work, i bought the version DFP-34G-2C2 instead of DFP-34X-2C2. But as i checked in the repo, the newer version like mine come with realtek chip, maybe RTL9602(i lost access now but i checked in past in /proc/cpuinfo). So i am using always the X version of firmware. Do you know if i will loose anything using the G version? (mine using realtek chip too)

**Reply by lehoangnb** [reply]
Hi, I dont have original .bin file, but my SFP come with 220923 firmware, I think you can use 220923 modded firmware without problem
G is gpon only mode, if your ISP using GPON standard, it will work.

**Reply by duskyzaoKING** [reply]
The problem is my config has a flash set SDS that I lost access in IP interface. When you flash the firmware they preserve this config, so I still without access. But I done UART and switched backed SDS and recovery IP access.

lost entire night trying to do UART, and the morning I checked I lost one capacitor and two resistor 🤦‍♂️.  As I have two units I checked the capacitors nF in other board and I am doing reverse engineering to find in one of my iPhone donate boards one capacitor and two resistors with same value. Already found the capacitor, just need to find resistor 🙏
![image](https://github.com/Anime4000/RTL960x/assets/1965743/84d9e5fc-b76d-47b6-988d-bdfdf6b63cae)

**Reply by Headcrabed** [reply]
Pull out and plug in fiber 5 times in 30 seconds would reset everything(except loid stuff)

**Reply by duskyzaoKING** [reply]
I didn’t know. But this erase SDS value too? The parameter to get 2.5GB

**Reply by Headcrabed** [reply]
My stock firmware has a option to change speed mode in web interface, and it works when that blocked me out.

**Reply by duskyzaoKING** [reply]
Trying to Finding this two micro resistors 😂😂 ![image](https://github.com/Anime4000/RTL960x/assets/1965743/c8c4b83c-23e6-4bcc-a1bc-3f899fb03cc2)

**Comment by duskyzaoKING** [comment]
@Anime4000 i flashed you BIN but they didn’t come with your interface customization. Maybe is it in the eprom? I only flashed the SPI

**Comment by Anime4000** [comment]
@lehoangnb I have question,
My stick using different brand SPI Flash, something off-brand called CFEON.
It should be similar to MXIC MX25L6433F right?

**Reply by duskyzaoKING** [reply]
You can check in datasheet of the two SPI Flash. But i can confirm its compatible, because i flashed your .bin and mine SPI its MXIC and worked ok. The bin didnt come with your logo customization, but came with your OMCI info and other things

**Reply by Anime4000** [reply]
the GUI Customization not yet public 😅 just to make sure no issue or glitch

**Reply by lehoangnb** [reply]
Maybe similar if both have same size, (but be aware, kernel in firmware must have driver for SPI flash)

**Reply by Anime4000** [reply]
Alright, just to be safe, read multiple time, backup multiple time, compare and verify multiple time, I'll get back once required tool arrive

**Comment by Anime4000** [comment]
I let everyone know,
`STICK3V08` do not have **24C04F EEPROM** thus cannot change SFP I2C

![Untitled](https://github.com/user-attachments/assets/ba4d7fde-a029-4728-86cf-f927bb00555a)
![Untitled-1](https://github.com/user-attachments/assets/d7edd3d8-7850-4dad-accc-c7d2fb384758)

but I found C2 Debugger pads
![Untitled](https://github.com/user-attachments/assets/290353a7-f043-4ec2-975e-a342b9cdd8dd)


I tried program SFP I2C `0x50` via CH341a I2C,...
with V06 stick, you can do that, but not V08 stick, some what have write protect...

**Reply by Headcrabed** [reply]
Wait, ODI has changed PCB design..?

**Comment by Anime4000** [comment]
A little update regarding to V08 stick I2C EEPROM, I have compiled @bigbigmdm SFP-Master (Linux only) to Windows

![CH341a](https://www.hitoha.moe/wp-content/uploads/2025/03/6275911921106535878.jpg)

![SFP-Master](https://www.hitoha.moe/wp-content/uploads/2025/03/Screenshot-2025-03-25-175850.png)
*which this need another driver that works with `libusb` Linux on Windows*

the V08 I2C has Write Protect, this piece of program can set password and write!
few user has verified, which this will make modifying SFP easier for OpenWRT or other SFP Host to work

all of this thanks to @lehoangnb make me believe stick can modify thus I going to find a solution to write SFP on the cheap

yea, I mod the CH341a Black to 3.3V

**Reply by bigbigmdm** [reply]
Hello @Anime4000 ! Can you send me a windows version of my IMSProg?

**Reply by Anime4000** [reply]
Hi @bigbigmdm Thank You very much your SFP-Master, which really help modifying GPON Sticks here
I have fork your SFP-Master and made some changes for cross compilation via MSYS2 (MinGW64)
https://github.com/Anime4000/SFP-Master/releases

I also detailed how to compile on Windows,

should I merge request? or leave Windows build to me?

**Reply by vuducdong** [reply]
> A little update regarding to V08 stick I2C EEPROM, I have compiled @bigbigmdm SFP-Master (Linux only) to Windows
> 
> ![CH341a](https://camo.githubusercontent.com/896ae6e1b0bf4173e125901c4cc6e50ae9a9245079a3749adda8ecbb950a0d5c/68747470733a2f2f7777772e6869746f68612e6d6f652f77702d636f6e74656e742f75706c6f6164732f323032352f30332f363237353931313932313130363533353837382e6a7067)
> 
> ![SFP-Master](https://camo.githubusercontent.com/f1348e5005713c4f91766993eb34dd69f156c259b0aeb3692e6b68d01d17f07c/68747470733a2f2f7777772e6869746f68612e6d6f652f77702d636f6e74656e742f75706c6f6164732f323032352f30332f53637265656e73686f742d323032352d30332d32352d3137353835302e706e67) _which this need another driver that works with `libusb` Linux on Windows_
> 
> the V08 I2C has Write Protect, this piece of program can set password and write! few user has verified, which this will make modifying SFP easier for OpenWRT or other SFP Host to work
> 
> all of this thanks to @lehoangnb make me believe stick can modify thus I going to find a solution to write SFP on the cheap
> 
> yea, I mod the CH341a Black to 3.3V

Bro, this game is fun, how to buy CH341A SFP Adapter? :D

**Reply by Anime4000** [reply]
@lehoangnb that `CH341A SFP Adapter` (Gerber File) need to order custom PCB via JLCPCB or PCBWay, there is BOM (Bill of Material) can order at difference place, and need self solder,

last time I order PCB MOQ is 5, I only need one

![xnNW0Y](https://github.com/user-attachments/assets/d479129c-ae2a-461a-bd60-d89a3feabc2a)

**Reply by CpServiceSpb** [reply]
Where can I buy a progranmer and adaptor boards ?

**Reply by bigbigmdm** [reply]
You can buy the programmer [here](https://aliexpress.ru/item/33054397283.html) and you can ordering assembling the adapter [here](https://easyeda.com/editor#id=00b5d14bad0e4a29a9d7101795f056e4) in `manufacturing`menu item.

**Comment by bigbigmdm** [comment]
Thank you, @Anime4000  ! I don't want to merge  pull request. It might cause a conflict with the Linux file structure.  Please leave the Windows version in your branch. Can i place the link to your branch in my readme.md file?

**Reply by Anime4000** [reply]
@bigbigmdm sure thing 👍 I'll manage the Windows builds

**Comment by bigbigmdm** [comment]
@Anime4000  , would you be willing to port my [IMSProg](https://github.com/bigbigmdm/IMSProg) to the Windows platform as well?

**Reply by Anime4000** [reply]
@bigbigmdm sure, I'll try make Windows Port as well

**Reply by Anime4000** [reply]
@bigbigmdm I have ported your IMSProg to Windows!
And test Z110 2.5GbE ONU, I didn't do chip-off reading, just quick test and it works
https://github.com/Anime4000/IMSProg

![image](https://github.com/user-attachments/assets/9b7be64f-a1fd-4d66-9efe-88f5d88bf56a)
![6278538568485946429](https://github.com/user-attachments/assets/2916a4a3-5bd5-4de6-9e4b-508a34b1301c)

**Reply by bigbigmdm** [reply]
Thank you, @Anime4000 !!! 
It's very good work!

**Comment by Headcrabed** [comment]
@Anime4000 Do you have a detailed instruction on i2c operation for changing v08 stick eeprom? Such tool above should not work if I want to modify eeprom directly using my router i2c.

**Reply by Anime4000** [reply]
tool above for CH341a and his PCB that I ordered from JLCPCB
based on traces to SFP pin, there is `Tx Disable` need to pull via resistor and ground,
also V08 stick is password write protected, in combination with `Tx Disable`

**Comment by bigbigmdm** [comment]
@Anime4000 , Jumper J3 is TxEnable (SFP pin 3), connected to ground with R3. You can open it,

**Reply by Headcrabed** [reply]
So pulling tx disable is not actually needed for programming eeprom?

**Reply by Anime4000** [reply]
during testing, it was connected, maybe you can try.

Password can be find in V06 sticks due to static EEPROM, look at 0x17B - 0x17E

**Comment by bigbigmdm** [comment]
Can you please tell me if the OTP register of the MX6433F chip on your device is empty?

**Comment by CpServiceSpb** [comment]
Do you have full assembled kit on your stock ?
As I see you are from Russia.
I can buy in Russia.

ср, 18 июн. 2025 г. в 19:57, Mikhail Medvedev ***@***.***>:

**Comment by bigbigmdm** [comment]
I am a private person, developer of the program [SFP-master](https://github.com/bigbigmdm/SFP-Master), circuit and PCB of the adapter. I do not have a stock. But I have 2 - 3 assembled adapters, because in China you can order the production of at least 10 printed circuit boards.

**Reply by bigbigmdm** [reply]
The SFP-master is now a new device:
<img width="1055" height="571" alt="sfp-master-pcb" src="https://github.com/user-attachments/assets/e0ec88b8-3da0-471f-96e2-ce28cb648e8c" />
<img width="900" height="360" alt="sfp-master-case2" src="https://github.com/user-attachments/assets/d17fec24-f820-4f22-b433-3db2c75a3f0a" />

Please, see [here](https://oshwlab.com/einkreader/sfp-master)

**Comment by trdb33** [comment]
> # Its for ODI DFP-34x-2c2 with 8MB MX25L6433F flash and 24C04 EEPROM
> 
> ![PCB - Copy](https://private-user-images.githubusercontent.com/5148339/329430079-da21178b-03b4-4139-a91a-47f5a200f086.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3MTQwNzEsIm5iZiI6MTc1NTcxMzc3MSwicGF0aCI6Ii81MTQ4MzM5LzMyOTQzMDA3OS1kYTIxMTc4Yi0wM2I0LTQxMzktYTkxYS00N2Y1YTIwMGYwODYuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMTgxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGM2M2U3MzMyZDNhYjI5ZmQyZDFiOTRiOGY3YzFjNDZiMmI0NGNhMTY2YjY5ZjVmYTMyM2UyYjMwYmQyODc1NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ._7q2_jD-SxUIV_dkJSI9nWTikHQr6zZVgK6Tl622IDc)
> ## SPI Flash info
> 
>     * Flash name: MX25L6433F SPI 8MB
> 
>     * File name: ODI_DFP-34X-2C2_factory_8M_v220923.bin
> 
>     * sha256: 5e3572c2c6821b8f345d6564ae547aaf734b227c678744b276d311cb97006cd7
> 
>     * Both layout is running 220923 @Anime4000 modified firmware
> 
> 
> ![Flash_info](https://private-user-images.githubusercontent.com/5148339/329429892-77430201-e211-4733-87ab-d2e34c4692e2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3MTQwNzEsIm5iZiI6MTc1NTcxMzc3MSwicGF0aCI6Ii81MTQ4MzM5LzMyOTQyOTg5Mi03NzQzMDIwMS1lMjExLTQ3MzMtODdhYi1kMmUzNGM0NjkyZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMTgxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGEyYmU1MWI5ZWY5NjgxZTZjOTU5MmUxNjNmZjA2ZGQ2NzhmMzUxM2M4YzNhYmQ0MGZmNzVmYjFiY2FiMWMxMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.n99nehniZiEn7c5KeApvcKLrtcqCZb5F0JOC1bFV4js) ![MX25L6433F_Pinout](https://private-user-images.githubusercontent.com/5148339/329429896-c0b8e5af-ef7a-46bf-977d-ee72b1f6d7cf.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3MTQwNzEsIm5iZiI6MTc1NTcxMzc3MSwicGF0aCI6Ii81MTQ4MzM5LzMyOTQyOTg5Ni1jMGI4ZTVhZi1lZjdhLTQ2YmYtOTc3ZC1lZTcyYjFmNmQ3Y2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMTgxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2FmMjA2MTIzMjhjNGUwMWRlNzI3MWEwNjdmMDliOTJkMjIzNmVmNGFjYThkYmYzYmUyYWM4ZWZlZDU3MjVkYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.r53AiE2x_TAPO7IVh6k5CBkjCzt6C79UmUpyJ8tM7tc) ![Test Clips Socket - Copy](https://private-user-images.githubusercontent.com/5148339/329430129-52a13b9a-418d-4679-9100-4ba45cffe67a.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3MTQwNzEsIm5iZiI6MTc1NTcxMzc3MSwicGF0aCI6Ii81MTQ4MzM5LzMyOTQzMDEyOS01MmExM2I5YS00MThkLTQ2NzktOTEwMC00YmE0NWNmZmU2N2EuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMTgxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzAwYjIyM2RkODIwM2MzMzk5Y2JjNTcyYTkwYzRkMGU4OGJmMjI0ODg0OWFlM2Y3NDYwYjY1NDMwZTZiMjkyNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.qn1eQTGZ80MUnH1Mt2OX7Zn2oyD-rCyGcCJiPcvQilQ)
> ## EEPROM info
> 
>     * EEPROM name: 24C04 4K (512B) I2C (SOT23 package)
> 
>     * File name: ODI_DFP-34X-2C2_eeprom_4K.bin
> 
>     * sha256: 6342c4bdefee80e1ca238615c3840184eaec5cdae48dc403111bb5b09f96fef0
> 
> 
> ![EEPROM_INFO](https://private-user-images.githubusercontent.com/5148339/329429886-af31f9aa-3ec5-4c44-b956-8988ebaecf81.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3MTQwNzEsIm5iZiI6MTc1NTcxMzc3MSwicGF0aCI6Ii81MTQ4MzM5LzMyOTQyOTg4Ni1hZjMxZjlhYS0zZWM1LTRjNDQtYjk1Ni04OTg4ZWJhZWNmODEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMTgxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjEwNTFlYmYyMWUwY2M2MmNkMDcwNTNkNTFmY2MyMzgxYWRiZDllMDNhMzBjODZhZGJkNTNmYTEyZTJjZDI5YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.SA1Le4BR0MXYAo_HHToAUeVkNCs7rPzXgKGAFCBznwI) ![EEPROM_Program_wiring - Copy](https://private-user-images.githubusercontent.com/5148339/329431699-88d63b12-09cf-4e24-8e17-89e447e47217.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3MTQwNzEsIm5iZiI6MTc1NTcxMzc3MSwicGF0aCI6Ii81MTQ4MzM5LzMyOTQzMTY5OS04OGQ2M2IxMi0wOWNmLTRlMjQtOGUxNy04OWU0NDdlNDcyMTcuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMTgxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDhkMDU5MjdmYjAyYTAxMmQ0M2ZhOWY1ZGYyNzdlZmZhOTAxMjViMTA0MzcyODIxZDYxZDc3YWQxZjBmYWY4YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.G58xGeBTTVlJAqo3qX7RBgfmS6d1KxIOBT_ePRb5BcM) ![24C04-SOT23_pinout](https://private-user-images.githubusercontent.com/5148339/329431728-55a6e776-3b02-42f2-b286-b777abf566f9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3MTQwNzEsIm5iZiI6MTc1NTcxMzc3MSwicGF0aCI6Ii81MTQ4MzM5LzMyOTQzMTcyOC01NWE2ZTc3Ni0zYjAyLTQyZjItYjI4Ni1iNzc3YWJmNTY2ZjkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMTgxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmE1MDExOTRmMTIxYWY2YjM1YzYyY2NlN2YxMGNjYzNkZGEzOGViZTRlYjg1YjUzMjYwYWY2YWI0N2RjZGQzMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FsWSZCNhjrpLuGPXAHQqukGOsYmOQCsS9JzM2qBcQ7s)
> ## Notice
> 
>     * Use only **3.3V** level for both Flash and EEPROM
> 
>     * In most case, flashing eeprom is not necessary
> 
> 
> ## Download
> 
> [ODI_DFP-34X-2C2_full_dump_v220923.zip](https://github.com/Anime4000/RTL960x/files/15269394/ODI_DFP-34X-2C2_full_dump_v220923.zip)

thanks, I finally reflashed my dongle and now it's working

**Comment by CpServiceSpb** [comment]
Is there the same for 16 Mb flash version ?

ср, 20 авг. 2025 г. в 21:17, trdb33 ***@***.***>:

**Comment by xx96m** [comment]
Hello dear friends, I have a question: after updating, I no longer have the automatic PPPOE account obtained by OLT. Why is there a configuration to do?
![mode router_1](https://github.com/user-attachments/assets/a4a179fe-45c0-4395-9a8d-a40144260068)
![mode router_new_1](https://github.com/user-attachments/assets/043efb71-16df-4ae7-9eb1-0cae3dabe3cb)
![WhatsApp Image 2025-08-25 at 14 15 13](https://github.com/user-attachments/assets/52cb840e-654f-4e20-8798-c3709f33077a)

**Comment by eLEGIJA** [comment]
<img width="1382" height="1245" alt="image" src="https://github.com/user-attachments/assets/df88df6a-666f-4666-ab9c-2a69069c3d16" />
Reading EEPROM by CH341A (black) in neoprogrammer. can't write .bin from here. what to do, how to unbrick odi-34x-2c2?

**Reply by eLEGIJA** [reply]
<img width="194" height="31" alt="image" src="https://github.com/user-attachments/assets/77e3be4b-8c45-4587-b228-3879d57d1a41" />
